python/bifrost/units.py

```diff
@@ -31,13 +31,16 @@
 ureg = pint.UnitRegistry()
 
 def convert_units(value, old_units, new_units):
-	old_quantity = value * ureg.parse_expression(old_units)
-	try:
-		new_quantity = old_quantity.to(new_units)
-	except pint.DimensionalityError:
-		raise ValueError("Cannot convert units %s to %s" %
+    old_quantity = value * ureg.parse_expression(old_units)
+    try:
+        new_quantity = old_quantity.to(new_units)
+    except pint.DimensionalityError:
+        raise ValueError("Cannot convert units %s to %s" %
 		                 (old_units, new_units))
-	return new_quantity.magnitude
+    if value >= 0:
+        return new_quantity.magnitude
+    else:
+        return -1 * new_quantity.magnitude
```

Comment:
Hmm, well, I only added it because my pipeline was crashing... I guess it depends on whether the `pint` library is returning the mathematical definition of magnitude, or something else (which actually seems more likely what the authors intended!)